src/Umbraco.Web/Security/BackOfficeUserStore.cs

```diff
@@ -245,25 +286,29 @@ public Task<bool> HasPasswordAsync(BackOfficeIdentityUser user)
         /// Set the user email
         /// </summary>
         /// <param name="user"/><param name="email"/>
+        /// <param name="cancellationToken"></param>
         /// <returns/>
-        public Task SetEmailAsync(BackOfficeIdentityUser user, string email)
+        public Task SetEmailAsync(BackOfficeIdentityUser user, string email, CancellationToken cancellationToken = default(CancellationToken))
         {
+            cancellationToken.ThrowIfCancellationRequested();
             ThrowIfDisposed();
             if (user == null) throw new ArgumentNullException(nameof(user));
-            if (email.IsNullOrWhiteSpace()) throw new ArgumentNullException("email");
+            // if (email.IsNullOrWhiteSpace()) throw new ArgumentNullException("email");
```

Comment:
do we need this check? else we can remove the comment

Reply:
I forgot to mention this one. Due to the Umbraco bootstrap process, a user with a null email address is created (it's later updated). What's Umbraco's policy with email? Is it required? Have I managed to change the bootstrap user creation somehow?

Reply:
@scottbrady91  This is a bit of a legacy story I suppose. Today, email is the login so it's required. But there are 2 fields in the db: `userLogin` and `userEmail`. There's a config switch somewhere IIRC that forces the username to be the email but if this is turned off they act independently. So by default, email cannot be blank. I guess I'm unsure where you are referring to where an empty email is set, have any pointers?

Reply:
During `NewInstallStep.ExecuteAsync`, a user is retrieved from the database that has an empty email address. This then fails validation during the `UserManager`s `ChangePasswordWithResetAsync` method.

Are you happy creating a new task for investigating this and leaving the check disabled for this PR?

Here's the callstack:
```
Umbraco.Web.Security.BackOfficeUserStore.SetEmailAsync(Umbraco.Web.Models.Identity.BackOfficeIdentityUser user, string email, System.Threading.CancellationToken cancellationToken) Line 296	C#
Umbraco.Web.Security.BackOfficeUserStore.SetNormalizedEmailAsync(Umbraco.Web.Models.Identity.BackOfficeIdentityUser user, string normalizedEmail, System.Threading.CancellationToken cancellationToken) Line 372	C#
[External Code]	
Umbraco.Web.Security.BackOfficeUserManager<Umbraco.Web.Models.Identity.BackOfficeIdentityUser>.ChangePasswordWithResetAsync(int userId, string token, string newPassword) Line 311	C#
Umbraco.Web.Install.InstallSteps.NewInstallStep.ExecuteAsync(Umbraco.Web.Install.Models.UserModel user) Line 73	C#
[External Code]	
Umbraco.Web.Install.Controllers.InstallApiController.ExecuteStepAsync(Umbraco.Web.Install.Models.InstallSetupStep step, object instruction) Line 256	C#
Umbraco.Web.Install.Controllers.InstallApiController.PostPerformInstall(Umbraco.Web.Install.Models.InstallInstructions installModel) Line 106	C#
[External Code]
```

Reply:
Alright, the issue seems to be because the superadmin (`UserId = -1`) has `email = ""`. This user is inserted into the database in `DatabaseDataCreator`

Reply:
@scottbrady91, as far as I can see, we can just move the following code in `NewInstallStep.ExecuteAsync`:
```
            admin.Email = user.Email.Trim();
            admin.Name = user.Name.Trim();
            admin.Username = user.Email.Trim();

            _userService.Save(admin);
```

up before the password is reset. Then the user has an email when `ChangePasswordWithResetAsync` is called

Reply:
I can do this change, and then we can merge :)

Reply:
Thanks, Bjarke